internal/controller/thanosquery_controller.go

```diff
@@ -214,10 +220,12 @@ func (r *ThanosQueryReconciler) getStoreAPIServiceEndpoints(ctx context.Context,
 		client.InNamespace(query.Namespace),
 	}
 	if err := r.List(ctx, services, listOpts...); err != nil {
+		r.Recorder.Event(&query, corev1.EventTypeWarning, "EndpointDiscoveryFailed", fmt.Sprintf("Failed to list StoreAPI services: %v", err))
```

Comment:
this is a transient error that will trigger a reconcile. id remove it to avoid noise